libraries/GCS_MAVLink/GCS.h

```diff
@@ -108,6 +110,173 @@ enum ap_message : uint8_t {
     }
 #define MAV_STREAM_TERMINATOR { (streams)0, nullptr, 0 }
 
+class MissionItemProtocol
+{
+public:
+    virtual MAV_MISSION_TYPE mission_type() const = 0;
+    void handle_mission_count(class GCS_MAVLINK *link,
+                              const mavlink_mission_count_t &packet,
+                              const mavlink_message_t &msg);
+    virtual void _handle_mission_count(class GCS_MAVLINK *link,
+                                       const mavlink_mission_count_t &packet,
+                                       const mavlink_message_t &msg) = 0;
+
+    void handle_mission_request_list(const GCS_MAVLINK *link,
+                                     const mavlink_mission_request_list_t &packet,
+                                     const mavlink_message_t &msg);
+
+    void handle_mission_write_partial_list(GCS_MAVLINK *link,
+                                           const mavlink_message_t &msg,
+                                           const mavlink_mission_write_partial_list_t &packet);
+    void handle_mission_item(const mavlink_message_t &msg,
+                             AP_Mission::Mission_Command &cmd);
+
+    void queued_request_send();
+
+    bool active_link_is(const GCS_MAVLINK *_link) const { return _link == link; };
+
+    void update();
+
+    bool            receiving; // currently receiving
+    uint16_t        request_i; // request index
+    uint16_t        request_last; // last request index
+
+    // waypoints
+    uint8_t        dest_sysid; // where to send requests
+    uint8_t        dest_compid; // "
+    uint32_t        timelast_receive; // milliseconds
+    uint32_t        timelast_request; // milliseconds
+    const uint16_t  receive_timeout = 8000; // milliseconds
```

Comment:
Rename to include `_ms` at the end for clarity? I actually thought this was seconds by the name orginally.

Reply:
Yep.